libbeat/outputs/otelconsumer/otelconsumer.go

```diff
@@ -97,10 +97,24 @@ func (out *otelConsumer) logsPublish(_ context.Context, batch publisher.Batch) e
 		pcommonEvent.CopyTo(logRecord.Body().SetEmptyMap())
 	}
 
-	if err := out.logsConsumer.ConsumeLogs(context.TODO(), pLogs); err != nil {
-		return fmt.Errorf("error otel log consumer: %w", err)
+	err := out.logsConsumer.ConsumeLogs(ctx, pLogs)
+	if err != nil {
+		// Permanent errors shouldn't be retried. This tipically means
+		// the data cannot be serialized by the exporter that is attached
+		// to the pipeline or when the destination refuses the data because
+		// it cannot decode it. Retrying in this case is useless.
+		//
+		// See https://github.com/open-telemetry/opentelemetry-collector/blob/1c47d89/receiver/doc.go#L23-L40
+		if consumererror.IsPermanent(err) {
+			batch.Drop()
```

Comment:
I don't think dropping the batch is a good idea, but what else we can do in this case?

Reply:
Drop is reasonable, we behave in a similar way in kafka and logstash beats output.  We do need to increment `st.PermanentErrors` and log the error.

Interestingly, the elasticsearch exporter doesn't have any instances of `consumererror.NewPermanent(err)`, so today all errors would be re-tryable.  But that could change in the future.  So this means our contract on what gets retried is controlled by the otel exporter that is configured.  And that contract could change as the exporter gets updated.

kafka and fileexporter do use `NewPermanent`, and interestingly so does opensearchexporter.

Reply:
Drop is fine, we do this in other outputs.